src/mbgl/map/transform.cpp

```diff
@@ -243,8 +243,8 @@ void Transform::flyTo(const CameraOptions &camera, const AnimationOptions &anima
 
     // When u₀ = u₁, the optimal path doesn’t require both ascent and descent.
     bool isClose = std::abs(u1) < 0.000001;
-    // Perform a more or less instantaneous transition if the path is too short.
-    if (isClose && std::abs(w0 - w1) < 0.000001) {
+    // Perform a more or less instantaneous transition if the path is too short or zoom out factor is inf.
+    if ((isClose && std::abs(w0 - w1) < 0.000001) || isinf(r(0))) {
```

Comment:
@1ec5 good call on descent factor, let me try reproducing the reverse interaction. 

`What causes the ascent factor to be infinite`: a division by 0 in the following code

```
    auto r = [=](double i) {
        /// bᵢ
        double b = (w1 * w1 - w0 * w0 + (i ? -1 : 1) * rho2 * rho2 * u1 * u1) / (2 * (i ? w1 : w0) * rho2 * u1);
        return std::log(std::sqrt(b * b + 1) - b);
    };
```